### ISSUE TITLE
Error when trying to build a plugin, *Server does not implement caddy.GracefulServer 

### ISSUE THREAD

**Issue by spl0i7** [coredns/coredns#2421]
I was following this [guide](https://coredns.io/2017/03/01/how-to-add-plugins-to-coredns/) to write a plugin core dns but I cannot generate the binary. I also tried to pretty much copy the  `whoami` but when I try to generate the binary, I get the following error
```
~/G/c/s/g/c/c/p/whoami ❯❯❯ go build; and go generate
# github.com/coredns/coredns/core/dnsserver
../../core/dnsserver/register.go:252:5: cannot use new(Server) (type *Server) as type caddy.GracefulServer in assignment:
        *Server does not implement caddy.GracefulServer (missing WrapListener method)
```

How do I fix this so that I can generate binary for my plugin which can be used with coredns?

**Comment by spl0i7** [comment]
oh I think I need to build the whole binary of coredns again to make plugin available.

**Comment by miekg** [comment]
No. You need a specific version of caddy, although we may be able to
upgrade the one in coredns as well. See the Makefile for the version we use.

**Comment by spl0i7** [comment]
thank you very much!

**Comment by Xuanwo** [comment]
related to https://github.com/coredns/coredns/pull/2505

**Comment by lllama** [comment]
Link to the appropriate line in the Makefile for the lazy:
https://github.com/coredns/coredns/blob/master/Makefile#L34

**Comment by Xuanwo** [comment]
This should be fixed in https://github.com/coredns/coredns/pull/2541 .

**Comment by yongtang** [comment]
I will close this issue as fix has been merged in #2541. But feel free to reopen.
